### PR TITLE
Automatically unlabel 'safe to test' after tests run

### PR DESCRIPTION
**Pull request by talarian1**
- [x] All [tests](https://github.com/jfrog/jfrog-cli#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] All [static analysis checks](https://github.com/jfrog/jfrog-cli/actions/workflows/analysis.yml) passed.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
-----

### CODE REVIEW

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -14,14 +14,19 @@ jobs:
     if: contains(github.event.pull_request.labels.*.name, 'safe to test') || github.event_name == 'push'
     strategy:
       fail-fast: false
```
Comment (yahavi) [inline .github/workflows/tests.yml:16]: This not needed if you're using only 1 agent
```suggestion
```

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -183,6 +190,7 @@ jobs:
         run: go test -v -timeout 0 --test.docker=true --jfrog.url=${{ secrets.PLATFORM_URL }} --jfrog.adminToken=${{ secrets.PLATFORM_ADMIN_TOKEN }} --rt.dockerRepoDomain=${{ secrets.CONTAINER_REGISTRY }} --rt.dockerVirtualRepo=${{ secrets.DOCKER_VIRTUAL }} --rt.dockerLocalRepo=${{ secrets.DOCKER_LOCAL }} --rt.dockerRemoteRepo=${{ secrets.DOCKER_REMOTE }} --rt.dockerPromoteLocalRepo=${{ secrets.DOCKER_PROMOTE_LOCAL }}
 
   Scripts-tests:
+    needs: Go-Lint
```
Comment (yahavi) [inline .github/workflows/tests.yml:193]: This job can safely run without any restrictions
```suggestion
```

**Review by yahavi — CHANGES_REQUESTED** [review]
Let's also rename Go-Lint to something else. I suggest `Pretest`

**Review by eyalbe4 — CHANGES_REQUESTED** [review]
I noticed the following files were removed.
I assume this is because they are unused. Just making sure.
image

**Comment by talarian1** [comment]
@eyalbe4, worse than that. 
Those files, cause the flowing messages to appear :

> 08:28:49 [Warn] The file /Users/yahavi/.nuget/packages/system.reflection.metadata/1.6.0/system.reflection.metadata.1.6.0.nupkg doesn't exist in the NuGet cache directory but it does exist as a target in the assets files. Skipping adding this dependency to the build info. This might be because the package already exists in a different NuGet cache, possibly the SDK's NuGetFallbackFolder cache. Removing the package from this cache may resolve the issue. 

And fails the `TestXrayAuditNugetMultiProject`

**Pull request merged**